xija/component/heat.py

```diff
@@ -640,6 +654,21 @@ def dvals(self):
             else:
                 self.calc_earth_vis_from_taco(ephems, q_atts)
 
+            # This next bit optionally checks to see if the solar ephemeris
+            # was passed in, and if it was it computes the fraction of the
+            # Earth's surface that is illuminated by the Sun. 
+            solar_xyzs = [getattr(self, 'solarephem0_{}'.format(x))
+                          for x in ('x', 'y', 'z')]
+
+            if not None in solar_xyzs:
+
+                solars = np.array([x.dvals for x in solar_xyzs]).transpose().copy()
+
+                cos = np.sum(ephems*solars, axis=1)
+                es = np.sum(ephems*ephems, axis=1)*np.sum(solars*solars, axis=1)
+                cos /= np.sqrt(es)
+                self._dvals *= 0.5*(1.0+cos)
```

Comment:
Should there be a convenient way to get access to this multiplicative factor via plotting or at least as an attribute to allow for diagnostics? I don't remember enough of the details to know how hard this would be.